src/monster_automaton.cpp

```diff
@@ -1131,7 +1235,15 @@ void Entity::automatonRecycleItem()
 	{
 		return;
 	}
+	
+	if ( this->monsterSpecialState == 0 )
```

Comment:
All of this `this` is redundant, usually shouldn't be used.
Should phase out use of `this` in member functions and just access the members directly.